cpp/src/arrow/flight/sql/odbc/entry_points.cc

```diff
@@ -51,6 +51,14 @@ SQLRETURN SQL_API SQLFreeConnect(SQLHDBC conn) {
   return arrow::SQLFreeHandle(SQL_HANDLE_DBC, conn);
 }
 
+SQLRETURN SQL_API SQLGetDiagField(SQLSMALLINT handleType, SQLHANDLE handle,
```

Comment:
Please use `SQLGetDiagFieldW` (and same thing for `SQLGetDiagRecW`) to specify that the driver is using unicode version, this is related to my changes in https://github.com/Bit-Quill/arrow/pull/32, can chat tmr during standup if needed

```suggestion
SQLRETURN SQL_API SQLGetDiagFieldW(SQLSMALLINT handleType, SQLHANDLE handle,
```

Reply:
Changed to use W semantic to indicate use of unicode.